packages/grafana-data/src/transformations/transformers/order.ts

```diff
@@ -39,15 +39,21 @@ export const createOrderFieldsComparer = (indexByName: Record<string, number>) =
   return indexOfField(a, indexByName) - indexOfField(b, indexByName);
 };
 
-const createFieldsOrderer = (indexByName: Record<string, number>) => (fields: Field[]) => {
+const createFieldsOrderer = (indexByName: Record<string, number>) => (
+  fields: Field[],
+  data: DataFrame[],
+  frame: DataFrame
+) => {
   if (!Array.isArray(fields) || fields.length === 0) {
     return fields;
   }
   if (!indexByName || Object.keys(indexByName).length === 0) {
     return fields;
   }
   const comparer = createOrderFieldsComparer(indexByName);
-  return fields.sort((a, b) => comparer(a.name, b.name));
+  return fields.sort((a, b) =>
+    comparer(calculateFieldDisplayName(a, frame, data), calculateFieldDisplayName(b, frame, data))
```

Comment:
was there a problem using getFieldDisplayName?

Reply:
no, just the `calculateFieldDisplayName` is a leaner one. This function stays private to grafana-data, but we can use the `getDisplayName` too and revert this change.

Reply:
getFieldDisplayName is the leaner one (it includes caching), and should be used always, afraid if calculateFieldDisplayName get's exported it might see some usage as well